### PR TITLE
Ccf 11021 validate configure

### PR DESCRIPTION
**Pull request by jgielstra-cs**
# Checklist:
Please check you fulfill ALL of the relevant checkboxes
- [ ] Notified docs of any potential USER-facing changes
- [ ] Added short description of the change - with relevant motivation and context. 
- [ ] Branch has the ticket number in its name (along with a ticket summary)
- [ ] Commented the code, particularly in hard-to-understand areas
- [ ] Added tests that prove my fix is effective or that my feature works
- [ ] Ran `npm test` and it passes
- [ ] Changes generate no new warnings

### CODE REVIEW

File: src/commands/configure.js [hunk 1]
```diff
@@ -24,13 +24,28 @@ const { readConfig, defaultConfig } = require('../config');
 const { printSuccess, printError } = require('./utils');
 
 const DEFAULT_CORTEX_URL = 'https://api.cortex.insights.ai';
+const URL_REGEX=/^(https?):\/\/(-\.)?([^\s/?\.#-]+\.?)+(\/[^\s]*)?$/;
```
Comment (clee-cs) [inline src/commands/configure.js]: I'm a bit rusty with my regexes but shouldn't it be `http` instead of `https`?

Comment (jgielstra-cs) [reply src/commands/configure.js]: http with an optional s...

Comment (clee-cs) [reply src/commands/configure.js]: my bad :)

File: src/commands/configure.js [hunk 2]
```diff
@@ -24,13 +24,28 @@ const { readConfig, defaultConfig } = require('../config');
 const { printSuccess, printError } = require('./utils');
 
 const DEFAULT_CORTEX_URL = 'https://api.cortex.insights.ai';
```
[the remark this answers is not in the document]
Comment (jgielstra-cs) [reply src/commands/configure.js]: Yea Added.. moved to it suppress URL prompt..

File: src/commands/configure.js (hunk 1 above)
Comment (jgielstra-cs) [reply src/commands/configure.js]: Nah kind of a strange way to accomplish it ..

File: src/commands/configure.js [hunk 3]
```diff
@@ -51,14 +66,16 @@ module.exports.ConfigureCommand = class {
 
             console.log(`Configuring profile ${chalk.green.bold(profileName)}:`);
             let cortexUrl = (profileUrl) ? profileUrl : yield prompt(`Cortex URL [${defaultCortexUrl}]: `);
+            // strip any trailing /
+            const URL_VALIDATION_MSG=`Invalid url "${cortexUrl}" expecting (http[s]://<hostname>)`;
```
Comment (smichalski-cs) [inline src/commands/configure.js]: Please change message to read: "Please enter a valid URL (http[s]://<hostname>)"

File: src/commands/configure.js [hunk 4]
```diff
@@ -51,14 +66,16 @@ module.exports.ConfigureCommand = class {
...
+            cortexUrl = _validate((cortexUrl || defaultCortexUrl),URL_VALIDATION_MSG ,URL_REGEX);
+            cortexUrl =cortexUrl.replace(/\/$/, '');
             let account = (profileAccount) ? profileAccount : yield prompt(`Account [${defaultAccount}]: `);
+            account = _validate(account || defaultAccount, 'Invalid account name, expecting non-empty value.');
```
Comment (smichalski-cs) [inline src/commands/configure.js]: Change message to read: "An account name is required."

File: src/commands/configure.js [hunk 5]
```diff
@@ -51,14 +66,16 @@ module.exports.ConfigureCommand = class {
...
             let username = (profileUsername) ? profileUsername : yield prompt(`Username [${defaultUsername}]: `);
-            const password = (profilePassword) ? profilePassword : yield prompt.password('Password: ');
-
-            cortexUrl = cortexUrl || defaultCortexUrl;
-            cortexUrl = cortexUrl.replace(/\/$/, ''); // strip any trailing /
-            account = account || defaultAccount;
-            username = username || defaultUsername;
+            username = _validate(username || defaultUsername, 'Invalid user name, expecting non-empty value.');
```
Comment (smichalski-cs) [inline src/commands/configure.js]: Change message to read: "A user name is required."

File: src/commands/configure.js [hunk 6]
```diff
@@ -51,14 +66,16 @@ module.exports.ConfigureCommand = class {
...
+            const password = _validate((profilePassword) ? profilePassword : yield prompt.password('Password: '),
+                'Invalid password, expecting non-empty value.');
```
Comment (smichalski-cs) [inline src/commands/configure.js]: Change message to read: "A password is required."

**Pull request merged**